### DISCUSSION TITLE
2 in 1 Devices - Automatically switch to fullscreen start menu (Windows 10 style)

### DISCUSSION THREAD

**Discussion by Independent-Impala** [valinet/ExplorerPatcher#978] [Ideas]
First of all many thanks for your amazing work so far! You have made me fear the downgrade in customizability way less that comes with Windows 11. With your contribution the actual version of the OS will make less of an impact when it comes to individual preference in the overall user experience. 

Now, I have one remaining wish for the Windows 10 style start menu that if fulfilled would make this perfect in my eyes. 

I own a Surface Pro and have set it to automatically switch to the fullscreen Start Menu once I detach my TypeCover. Would it be possible to enable this also? This feature is a must have in my opinion for anybody who owns a 2 in 1 device!

Either way, thank you again for your hard work thus far and I sincerely hope this feature will get implemented in the future!

Best wishes,

Independent Impala

**Comment by sdw4938** [comment]
This sounds like a great idea!

**Comment by valinet** [comment]
What makes a device a 2-in-1? If I unplug my keyboard from my desktop PC, should it switch to full screen...? Idk, the thing has to be studied, probably there is some API or something undocumented at least that Microsoft uses to determine this. I don't know of it, it has to be researched.

Also, a slight inconvenience when implementing this is the fact that toggling the policy that makes Start full screen or menu requires admin, while the whole application runs as standard user. If implemented, a workaround this would have to be devised.

Yeah, maybe, idk. In the meantime, playing with something like Task Scheduler and triggering when the device switches between modes, if possible, would be your best bet. Actually, kind of simpler, if possible. When you trap the event, just set the registry entry corresponding to the [`ForceStartSize` policy](https://admx.help/?Category=Windows_10_2016&Policy=Microsoft.Policies.StartMenu::ForceStartSize). The Windows 10 Start menu hooks installed by EP catch the modification and apply it, so setting it in the registry is enough to trigger the change.

**Comment by ghost** [comment]
Hi :) 

Just a note that Windows 11 by default don't have a tablet mode of what you speak, all it does now is expand the icons more far apart and there is a new nicer layout for tablet users to that is in dev mode right now.

So, you are asking that when you have keyboard on tablet-like Surface pro, it keeps the default non-full screen apps then when switch to tablet mode, you want it to go to full screen?
